File: haskell/src/Shapes/Base/Models/Square.hs
module Shapes.Base.Models.Square (Square (..), newSquare) where

import Shapes.Base.Commons

data Square = Square {side :: Double} deriving (Show, Eq)

instance Dimensional Square where
    area (Square side) = side * side
    perimeter (Square side) = side * 4

newSquare :: Double -> Square
newSquare side = Square (max side 1)

File: csharp/Src/Shapes/Base/Models/Square.cs
using Shapes.Base.Commons;

namespace Shapes.Base.Models;

public sealed class Square : IDimensional {
    private readonly double _side;

    public Square(double side) {
        double dim1 = side > 0 ? side : 1;
        _side = dim1;
    }

    public double Area() {
        return Math.Pow(_side, 2);
    }

    public double Perimeter() {
        return 4 * _side;
    }
}
